### ISSUE TITLE
[tune] why set hyperparam both config and scheduler

### ISSUE THREAD

**Issue by MrRace** [ray-project/ray#11000]
I use the master example  from `ray/python/ray/tune/examples/pbt_transformers/pbt_transformers.py`, 

1.  I find in the `config` and scheduler's hyperparam_mutations contain the `weight_decay, learning_rate, per_gpu_train_batch_size` . Why set the parameter in both `config` and `scheduler`, is it requisite? 

```
 config = {
        "model_name": model_name,
        "task_name": task_name,
        "data_dir": task_data_dir,
        "per_gpu_val_batch_size": 32,
        "per_gpu_train_batch_size": tune.choice([16, 32, 64]),
        "learning_rate": tune.uniform(1e-5, 5e-5),
        "weight_decay": tune.uniform(0.0, 0.3),
        "num_epochs": tune.choice([2, 3, 4, 5]),
        "max_steps": 1 if smoke_test else -1,  # Used for smoke test.
    }

    scheduler = PopulationBasedTraining(
        time_attr="training_iteration",
        metric="eval_acc",
        mode="max",
        perturbation_interval=1,
        hyperparam_mutations={
            "weight_decay": lambda: tune.uniform(0.0, 0.3).func(None),
            "learning_rate": lambda: tune.uniform(1e-5, 5e-5).func(None),
            "per_gpu_train_batch_size": [16, 32, 64],
        })
```

2.  By the way,  I set `num_samples=8`, I think the `==Status==`  in terminal logs should appear 8 times intuitively, but actually 45 times. Therefore do it mean during trials, it try 45 times and keep 8 times?

3.  Another confusion is below:
![image](https://user-images.githubusercontent.com/10300313/94140547-f90e3580-fe9d-11ea-87c2-a2731118442a.png)
I do not  what are there mean in the figure above, in more details：
(1)How do the three directories come from and  What are the three directories include?  Why only three not four ?
```
2020-09-24_18-20-12eaeay5m9/
2020-09-24_18-24-50tqbtvwn3/
2020-09-24_18-25-47q62ycwl5/
```
For example what the `12eaeay5m9`  directory contains is below:
![image](https://user-images.githubusercontent.com/10300313/94141695-ba797a80-fe9f-11ea-8c11-5cc524bdb0f7.png)

(2) there are three  `pbt_policy_64adc_*` files that seem match the three directories above,  is there any relationship between them?

Pardon for my such many doubts.Best regards!

**Comment by richardliaw** [comment]
cc @amogkam

**Comment by amogkam** [comment]
@MrRace thanks for the questions!

1) The difference is that config specifies the initial search space, and hyperparam_mutations in PBT specifies the mutation search space. But having it in both places is no longer required in Ray 1.0. If a hyperparam is not in config, the search space in PBT hyperparam_mutations will be used as the initial search space. You're right, we should update the example to not have it in both places.
2) This is odd. If you are running only 8 samples, then status should be showing only 8 trials. Do you have a screenshot for this?
3)  Also not sure about the 3 extra folders being created. If you delete your ray_results and run just pbt_transformers.py, do you still see those folders getting created?
4) The pbt_policy files have no relation to the 3 directories. pbt_policy_{i} contains all the pbt perturbations for trial i. pbt_global.txt contains the pbt perturbations for all trials. The reason theres only 3 pbt_policy_i files is probably because only those 3 trials underwent any perturbations. 

Hope this helps!
